libraries/net/include/psibase/auto_timeout.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <algorithm>
+
+namespace psibase::net
+{
+   // Automatically adjusts a timeout based on the observed network latency.
+   // The timeout increases by a factor of 1.5 every time it expires,
+   // but is periodically reduced if the maximum latency observed recently
+   // is significantly lower.
+   template <typename Clock>
+   class auto_timeout
+   {
+     public:
+      explicit auto_timeout(typename Clock::duration initial_value) : _current(initial_value)
+      {
+         start();
+      }
+      void stop() {}
```

Comment:
`stop` is a no-op?

Reply:
It doesn't actually need to do anything. I mainly used it as a hook to put debugging code or breakpoints.